### PR TITLE
Add jenkinsfile for getDependency build

### PR DESCRIPTION
**Pull request by Mesbah-Alam**
Resolves https://github.com/AdoptOpenJDK/openjdk-tests/issues/689
Signed-off-by: Mesbah Alam <Mesbah_Alam@ca.ibm.com>

### CODE REVIEW

**Comment by Mesbah-Alam** [comment]
Contend of `test.getDependency` as was in job configuration: 

```
timeout(time: 1, unit: 'HOURS') {
    stage ('Get Dependencies') {
        node ('master') {
            timestamps {
                try {
                    cleanWs()
                    sh 'wget https://raw.githubusercontent.com/eclipse/openj9/master/test/TestConfig/scripts/tools/getDependencies.pl'
                    sh 'perl ./getDependencies.pl -path . -task default -os os.linux'
                    archiveArtifacts '*.jar, *.zip'
                } finally {
                    cleanWs()
                }
            }
        }
    }
}
```

**Comment by Mesbah-Alam** [comment]
Content of `systemtest.getDepedency` as was in job configuration: 

```
timeout(time: 8, unit: 'HOURS') {
    stage ('Get SystemTest Dependencies') {
        node ('master') {
            timestamps {
                try {
                    cleanWs()
                    sh 'curl -OLJks  "https://api.adoptopenjdk.net/v2/binary/nightly/openjdk8?openjdk_impl=hotspot&os=linux&arch=x64&release=latest&type=jdk"'
                    sh 'mkdir ${WORKSPACE}/j2sdk-image'
                    sh 'tar -xzf OpenJDK8U-jdk_x64_linux_hotspot*.gz -C ${WORKSPACE}/j2sdk-image --strip-components 1'
                    sh '${WORKSPACE}/j2sdk-image/jre/bin/java -version'
                    sh 'git clone git://github.com/AdoptOpenJDK/openjdk-systemtest'
                    sh 'ant -f ./openjdk-systemtest/openjdk.test.mauve/build.xml -Djava.home=${WORKSPACE}/j2sdk-image/jre -Dprereqs_root=${WORKSPACE}/systemtest_prereqs'
                    archiveArtifacts '**/systemtest_prereqs/mauve/mauve.jar'
                } finally {
                    cleanWs()
                }
            }
        }
    }
}
```

**Comment by Mesbah-Alam** [comment]
Note: this PR combines the content of job configuration of both `test.getDependency` and 'systemtest.getDependency` and puts them into a single Jenkinsfile - differentiated by a job parameter called `BUILD_TYPE` -- set in the jobs -- with default value "systemtest" for systemtest.getDependency, and "test" for test.getDependency.

**Comment by Mesbah-Alam** [comment]
PR tested by 
test.getDependency : https://ci.adoptopenjdk.net/view/all/job/test.getDependency/157/
systemtest.getDependency: https://ci.adoptopenjdk.net/view/all/job/systemtest.getDependency/57/

File: buildenv/jenkins/getDependency [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	def time_limit = 8
+	if(params.TIME_LIMIT) {
+		time_limit = params.TIME_LIMIT.toInteger()
+	}
+	timeout(time: time_limit, unit: 'HOURS') {
+		try {
+			if( params.BUILD_TYPE == "systemtest" ){
+			    sh 'curl -OLJks  "https://api.adoptopenjdk.net/v2/binary/nightly/openjdk8?openjdk_impl=hotspot&os=linux&arch=x64&release=latest&type=jdk"'
+                sh 'mkdir ${WORKSPACE}/j2sdk-image'
+                sh 'tar -xzf OpenJDK8U-jdk_x64_linux_hotspot*.gz -C ${WORKSPACE}/j2sdk-image --strip-components 1'
```
Comment (sophia-guo) [inline buildenv/jenkins/getDependency]: What's the purpose of this tar.gz?

Comment (Mesbah-Alam) [reply buildenv/jenkins/getDependency]: We need the SDK to compile the checked-out mauve source code.

File: buildenv/jenkins/getDependency [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                sh '${WORKSPACE}/j2sdk-image/jre/bin/java -version'
+                sh 'git clone git://github.com/AdoptOpenJDK/openjdk-systemtest'
+                sh 'ant -f ./openjdk-systemtest/openjdk.test.mauve/build.xml -Djava.home=${WORKSPACE}/j2sdk-image/jre -Dprereqs_root=${WORKSPACE}/systemtest_prereqs'
```
Comment (sophia-guo) [inline buildenv/jenkins/getDependency]: What's the requirement of `java.home` to build mauve.jar? If it's not necessarily the test against JAVA grabbing a stable JAVA( release not a nightly) may be better.

File: buildenv/jenkins/getDependency [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+def testBuild() {
+	def time_limit = 8
+	if(params.TIME_LIMIT) {
+		time_limit = params.TIME_LIMIT.toInteger()
+	}
+	timeout(time: time_limit, unit: 'HOURS') {
+		try {
+			if( params.BUILD_TYPE == "systemtest" ){
+			    sh 'curl -OLJks  "https://api.adoptopenjdk.net/v2/binary/nightly/openjdk8?openjdk_impl=hotspot&os=linux&arch=x64&release=latest&type=jdk"'
```
Comment (sophia-guo) [inline buildenv/jenkins/getDependency]: See the following comment

File: buildenv/jenkins/getDependency (hunk 2 above)
Comment (Mesbah-Alam) [reply buildenv/jenkins/getDependency]: java.home is required by the systemtest scripts. Yes. I agree. Can you share link to a stable sdk tar file?

**Pull request merged**